SalesforceCommerceCloud/pwa-kit
# [POC]Integrate a11y scan into E2E tests

# Description
This PR added E2E tests for a11y by using snapshot method provided by playwright. The reason we are using snapshots instead of excluding/ixcluding a11y rules are because current storefront still have known a11y errors which may be addressed in the future.
With snapshot method, we are capturing the current stage of a11y on each page on the E2E that we have in the project

## Why
We want to make sure we are not introduce any a11y bugs into current existing code. Although we have fixed most of our a11y in GUS, but we still have some minor errors that have no plan to get fixed. 
Using snapshot, we are capturing the current state of a11y on the storefront

## Pros
- We don't have to disable/ignore any a11y rules when we run the a11y scan on E2E
- Any new a11y error will get noticed since it will cause nightly run to fail

## Cons
- We will need to update snapshots if there is an acceptable change to current a11y state. Say, you fixed an a11y or you may have introduce a new a11y, but it is acceptable for now to move forwards. 
- The snapshots file will need to be committed in the codebase
- This will not run on PR checked on every commit, but will run on E2E tests. We may have discovered if we violate any a11y at least one nightly run



# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- (change1)

# How to Test-Drive This PR

- Check out code
- Change you config to match E2E instance
 ```
// template-retail-app/default.js
parameters: {
                clientId: '987fc116-d30c-4537-93cb-c2bd433c3b5a',
                organizationId: 'f_ecom_zzrf_002',
                shortCode: '8o7m175y',
                siteId: 'RefArch'
 }
```
- Change your DEFAULT_SEARCH_PARAM in constant.js
```
export const DEFAULT_SEARCH_PARAMS = {limit: 3, offset: 0, sort: 'best-matches', refine: []}

```
- Run `RETAIL_APP_HOME=http://localhost:3000 test:e2e` and make sure no tests are failing.
- Now change something in your code that can cause a11y.
- Re run the tests, you will notice the e2e tests will fail and complain the snapshot for a11y is no longer matched.

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
